### ISSUE TITLE
latest multi-channel additions crash [outlet~]

### ISSUE THREAD

**Issue by Ant1r** [pure-data/pure-data#1894]
Since 691e0f8c85d19d323a623a72e91734e3017ec505, simply creating an [outlet~] then switching DSP on crashes Pd, on Linux at least (tested on Ubuntu 20.04 and 16.04).

On a fresh clone of current master (35e924d2051ec7afede93676c97b2700d460c23b), gdb returns the following backtrace:

``` GDB
$ gdb --args ./bin/pd -nrt
(...)
Reading symbols from ./bin/pd...
(gdb) r
Starting program: /home/antoine/Documents/Pd/pure-data-2/bin/pd -nrt
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[Detaching after fork from child process 44764]
      --- open new patch, create outlet~, turn DSP on ---
Program received signal SIGSEGV, Segmentation fault.
0x0000555555594215 in signal_setchansout (sig=0x0, nchans=1) at d_ugen.c:567
567	    *sig = signal_new((*sig)->s_length, nchans, (*sig)->s_sr, 0);
(gdb) bt
#0  0x0000555555594215 in signal_setchansout (sig=0x0, nchans=1) at d_ugen.c:567
#1  0x00005555555bf3e0 in voutlet_dsp (sp=0x5555557726e0, x=0x5555557721c0) at g_io.c:483
#2  voutlet_dsp (x=0x5555557721c0, sp=0x5555557726e0) at g_io.c:472
#3  0x00005555555950fb in ugen_doit (dc=dc@entry=0x555555773cd0, u=u@entry=0x555555773d50) at d_ugen.c:927
#4  0x000055555559624c in ugen_done_graph (dc=dc@entry=0x555555773cd0) at d_ugen.c:1199
#5  0x00005555555a4047 in canvas_dodsp (x=x@entry=0x555555761b30, toplevel=toplevel@entry=1, sp=sp@entry=0x0) at g_canvas.c:1381
#6  0x00005555555a40df in canvas_start_dsp () at g_canvas.c:1400
#7  0x00005555555f028b in binbuf_eval (x=<optimized out>, target=0x5555556bf848 <glob_pdobject>, target@entry=0x0, argc=argc@entry=0, 
    argv=argv@entry=0x0) at m_binbuf.c:789
#8  0x00005555555fc278 in socketreceiver_read (fd=14, x=0x55555575e0f0) at s_inter.c:685
#9  socketreceiver_read (x=0x55555575e0f0, fd=14) at s_inter.c:617
#10 0x00005555555fb26c in sys_domicrosleep (microsec=microsec@entry=0) at s_inter.c:234
#11 0x00005555555fd163 in sys_pollgui () at s_inter.c:1022
#12 0x00005555555f8955 in m_pollingscheduler () at m_sched.c:351
#13 m_mainloop () at m_sched.c:432
#14 0x00007ffff7b15083 in __libc_start_main (main=0x55555556e5c0 <main>, argc=2, argv=0x7fffffffdfa8, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7fffffffdf98) at ../csu/libc-start.c:308
#15 0x000055555556e61e in _start ()
(gdb) 

```

**Comment by Ant1r** [comment]
This crash only happens when the [outlet~] is created in a toplevel patch (like when opening the "List of objects" patch).
This is fixed by #1895.

**Comment by millerpuckette** [comment]
Yep, happens to me too.  Will dig this up later today.

**Comment by umlaeute** [comment]
confirmed

**Comment by Spacechild1** [comment]
Fixed with 24b788d
